R/conditions.R

```diff
@@ -1,15 +1,23 @@
 #' Custom conditions for vctrs package
 #'
+#' @description
+#'
 #' These errors and warnings have custom classes and structures to make
 #' testing easier.
 #'
+#' By default, lossy casts are an error. Use
+#' `allow_lossy_cast()` to silence these errors and continue
+#' with the partial results. The lost values may be set to `NA` or to
+#' a lower value resolution, depending on the type of cast.
```

Comment:
I think you need a paragraph or two here about `maybe_lossy_cast()` — it either throws an error or returns a vector, if `allow_lossy_cast()` is used